open-api/rest-catalog-open-api.yaml

```diff
@@ -2692,7 +2692,15 @@ components:
           $ref: '#/components/schemas/Schema'
         last-column-id:
           type: integer
-          description: The highest assigned column ID for the table. This is used to ensure columns are always assigned an unused ID when evolving schemas. When omitted, it will be computed on the server side.
+          deprecated: true
+          description:
+            This optional field is **DEPRECATED for REMOVAL** since it more safe to handle this internally,
+            and shouldn't be exposed to the clients.
+
+            The highest assigned column ID for the table. This is used to ensure columns are always
+            assigned an unused ID when evolving schemas. When omitted, it will be computed on the server side.
+
+            Deprecated since Iceberg (Java) 1.8.0. The field will be removed from this spec in Iceberg (Java) 2.0.
```

Comment:
This spec has nothing to do with Java releases, so I don't like including them here. A major Java release has meaning for Java consumers, but not to implementations of this spec.

Reply:
I took this approach from the deprecation of `getTokens`, but I agree. I've removed it for now.